api/team_members.go

```diff
@@ -0,0 +1,139 @@
+//
+// Author:: Vatasha White (<vatasha.white@lacework.net>)
+// Copyright:: Copyright 2021, Lacework Inc.
+// License:: Apache License, Version 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package api
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+)
+
+type TeamMembersService struct {
+	client *Client
+}
+
+// NewTeamMember returns an instance of the Team Member struct
+//
+// Basic usage: Initialize a new TeamMember struct and then use the new instance to perform CRUD operations.
+//
+//   client, err := api.NewClient("account")
+//   if err != nil {
+//     return err
+//   }
+//
+//   teamMember := api.NewTeamMember(
+//		"FooBar",
+//		"1",
+//		api.TeamMemberProps{
+//		Company: "ACME Inc",
+//		FirstName: "Foo",
+//		LastName: "Bar"
+//      },
+//   },
+// )
+//
+// client.V2.TeamMembers.Create(teamMember)
+//
+func NewTeamMember(username, userEnabled string, props TeamMemberProps) (TeamMember, error) {
+	userEnabledEnum, err := strconv.Atoi(userEnabled)
+	if err != nil {
+		return TeamMember{}, err
+	}
+
+	if userEnabledEnum > 1 || userEnabledEnum < 0 {
+		return TeamMember{}, errors.New("userEnabled field must be 0 for disabled or 1 for enabled")
+	}
+
+	return TeamMember{
+		Props:       props,
+		UserEnabled: userEnabled,
+		UserName:    username,
+	}, nil
+}
+
+// List returns a list of team members
+func (svc *TeamMembersService) List() (res TeamMembersResponse, err error) {
+	err = svc.client.RequestDecoder("GET", apiV2TeamMembers, nil, &res)
+	return
+}
+
+// Create creates a single team member
+func (svc *TeamMembersService) Create(tm TeamMember) (res TeamMemberResponse, err error) {
+	err = svc.client.RequestEncoderDecoder("POST", apiV2TeamMembers, tm, &res)
+	return
+}
+
+// Delete deletes a single team member with the corresponding guid
+func (svc *TeamMembersService) Delete(guid string) error {
+	if guid == "" {
+		errors.New("please specify a guid")
+	}
+
+	return svc.client.RequestDecoder("DELETE", fmt.Sprintf(apiV2TeamMembersFromGUID, guid), nil, nil)
+}
+
+// Update updates a single team member with the corresponding guid
+func (svc *TeamMembersService) Update(tm TeamMember) (res TeamMemberResponse, err error) {
+	if tm.CustGuid == "" {
+		err = errors.New("please specify a guid")
+		return
+	}
+	err = svc.client.RequestEncoderDecoder("PATCH", fmt.Sprintf(apiV2TeamMembersFromGUID, tm.UserGuid), tm, &res)
```

Comment:
For updates, verify that you are able to send the `UserGuid` as part of the request payload, some endpoints complain about it since you are already injecting it via URL parameter.